crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -77,7 +81,14 @@ impl Solution {
             return Err(error::Solution::InvalidClearingPrices);
         }
 
-        // Apply protocol fees
+        // Apply protocol fees only if the drivers is set to handler the fees
+        if fee_handler != FeeHandler::Driver {
+            return Ok(Self {
+                trades: solution.trades,
+                ..solution
+            });
```

Comment:
No need to use destructuring if we don't change any fields.
```suggestion
            return Ok(solution);
```

Reply:
totally true, with the rush I overlook we return the same, thanks!